### PR TITLE
test: Add unit tests to Firewall resource model and helper functions

### PR DESCRIPTION
**Pull request by ykim-1**
## 📝 Description

Adding unit tests to Firewall resource_models and helper functions

## ✔️ How to Test

╰─➤  go test -v framework_models_unit_test.go framework_datasource.go framework_models.go framework_schema_datasource.go helpers.go resource.go schema_resource.go

**How do I run the relevant unit/integration tests?**

## 📷 Preview

**If applicable, include a screenshot or code snippet of this change. Otherwise, please remove this section.**

### CODE REVIEW

**Review by lgarber-akamai — COMMENTED** [review]
Looks good so far! 
Maybe we could tag test files to differentiate unit from integration tests like they do [here](http://peter.bourgon.org/go-in-production/#testing-and-validation), so unit and integration tests can be ran from different Makefile targets.

**Comment by ykim-1** [comment]
> Looks good so far! Maybe we could tag test files to differentiate unit from integration tests like they do [here](http://peter.bourgon.org/go-in-production/#testing-and-validation), so unit and integration tests can be ran from different Makefile targets.

Sure sounds good! It will make our test files have long names. e.g. `resource_integration_test.go`, `data_source_integration_test.go` but it will be much easier to write the workflow/makefile

File: linode/firewall/firewall_helpers_unit_test.go [hunk 1]
```diff
@@ -0,0 +1,215 @@
+//go:build unit
+
+package firewall
```
Comment (zliang-akamai) [inline linode/firewall/firewall_helpers_unit_test.go:3]: Should we use `firewall_test` package here (I am not sure)?

**Review by lgarber-akamai — APPROVED** [review]
The new `make unittest` target works as expected and all tests seem to be passing. Great work!

File: makefile [hunk 2]
```diff
@@ -48,14 +48,16 @@ testacc: fmtcheck
...
-	go test -v ./$(PKG_NAME) $(TESTARGS) -count $(ACCTEST_COUNT) -timeout $(ACCTEST_TIMEOUT) -parallel=$(ACCTEST_PARALLELISM) -ldflags="-X=github.com/linode/terraform-provider-linode/version.ProviderVersion=acc"
+	go test --tags=integration -v ./$(PKG_NAME) $(TESTARGS) -count $(ACCTEST_COUNT) -timeout $(ACCTEST_TIMEOUT) -parallel=$(ACCTEST_PARALLELISM) -ldflags="-X=github.com/linode/terraform-provider-linode/version.ProviderVersion=acc"
 
 smoketest: fmtcheck
 	TF_ACC=1 \
 	LINODE_API_VERSION="v4beta" \
 	RUN_LONG_TESTS=$(RUN_LONG_TESTS) \
 	go test -v -run smoke ./linode/... -count $(ACCTEST_COUNT) -timeout $(ACCTEST_TIMEOUT) -parallel=$(ACCTEST_PARALLELISM) -ldflags="-X=github.com/linode/terraform-provider-linode/version.ProviderVersion=acc"
 
+unittest:
```
Comment (lgarber-akamai) [inline makefile:59]: I think we can call this target from the PR GHA workflow: https://github.com/linode/terraform-provider-linode/blob/dev/.github/workflows/pull_request.yml#L19

File: linode/firewall/firewall_helpers_unit_test.go (hunk 1 above)
Comment (ykim-1) [reply linode/firewall/firewall_helpers_unit_test.go:3]: Either way works :)

**Review by yec-akamai — APPROVED** [review]
Great work!

**Pull request merged**